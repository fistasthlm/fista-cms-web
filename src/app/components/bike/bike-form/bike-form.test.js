import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import BikeForm from './bike-form';

jest.mock('utils/history', () => ({
    push: params => params,
}));

const setup = propOverrides => {
    const props = Object.assign({
        user: Map({
            instagram: 'peteholmberg'
        }),
        onSubmit: jest.fn(),
    }, propOverrides);

    return shallow(<BikeForm {...props} />);
};

describe('Render', () => {
    it('should render component', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });
});

describe('State', () => {
    it('should handle text input change', () => {
        const wrapper = setup();
        const mockEvent = {
            target: {
                name: 'title',
                value: 'dolan',
            },
        };

        wrapper.instance().handleTextInputChange(mockEvent);

        expect(wrapper.state().title).toEqual('dolan');
    });

    it('handle upload image result', () => {
        const wrapper = setup();

        const mockEvent = {
            filesUploaded: [
                {
                    url: 'hej.com',
                    filename: 'dolan.png'
                }
            ]
        };

        wrapper.instance().handleUploadImageResult(mockEvent);

        expect(wrapper.state().images).toEqual([{
            url: 'hej.com',
            name: 'dolan.png'
        }]);
    });

    it('should submit', () => {
        const wrapper = setup();
        wrapper.state().title = 'dolan';
        wrapper.state().images = [{ url: 'dolan.com', name: 'dolan.png' }];
        const mockEvent = {
            preventDefault: jest.fn(),
        };

        wrapper.instance().submit(mockEvent);

        expect(wrapper.instance().props.onSubmit).toHaveBeenCalled();
    });
});
