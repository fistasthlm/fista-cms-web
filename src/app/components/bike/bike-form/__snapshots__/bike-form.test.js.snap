// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<form
  className="add-bike__form"
  onSubmit={[Function]}
>
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="title"
    placeholder="Title for bike"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="frame"
    placeholder="Frame"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="fork"
    placeholder="Fork / Headset"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="cranks"
    placeholder="Cranks / BB"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="pedals"
    placeholder="Pedals"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="drivetrain"
    placeholder="Drivetrain / Cog / Chainring / Chain"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="handlebars"
    placeholder="Handlebars / Stem"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="saddle"
    placeholder="Saddle / Seatpost"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="frontWheel"
    placeholder="Front Wheel / Hub / Tire"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="rearWheel"
    placeholder="Rear Wheel / Hub / Tire"
    type="text"
    value=""
  />
  <BikeInput
    className="add-bike__input"
    handleChange={[Function]}
    name="instagram"
    placeholder="Instagram"
    type="text"
    value="peteholmberg"
  />
  <div
    className="add-bike__photos"
  >
    <t
      apikey="A3BcPUqFURlSDHWjF3UG1z"
      buttonClass="add-bike__add-photo"
      buttonText="Add images"
      cname={null}
      file={null}
      link={false}
      mode="pick"
      onError={[Function]}
      onSuccess={[Function]}
      options={
        Object {
          "accept": "image/*",
          "fromSources": Array [
            "local_file_system",
          ],
          "maxFiles": 5,
          "storeTo": Object {
            "location": "s3",
          },
        }
      }
      render={null}
      security={null}
    />
  </div>
  <button
    className="add-bike__submit add-bike__submit--disabled"
    type="submit"
  >
    Save bieeek
  </button>
</form>
`;
